build/stf-run-ci/tasks/main.yml

```diff
@@ -20,19 +20,20 @@
     msg: __deploy_from_bundles_enabled not currently supported with __local_build_enabled (but should be)
   when: __local_build_enabled | bool and __deploy_from_bundles_enabled | bool
 
-- name: Get the node hostnames
-  set_fact:
-    nodes: "{{ lookup('k8s', kind='nodes') }}"
+- name: Get the list of nodes
+  k8s_info:
+    kind: Node
+  register: node_info
 
-- name: Find out if we are using crc
+- name: Find out if we are using crc by looking at the node hostnames
   block:
     - name: Try finding out for cluster with more than 1 node
       set_fact:
-        is_crc: "{{ True if 'crc' in nodes[0].metadata.labels[\"kubernetes.io/hostname\"] else False }}"
+        is_crc: "{{ True if 'crc' in node_info.resources[0].metadata.labels[\"kubernetes.io/hostname\"] else False }}"
   rescue:
     - name: Try finding out for cluster with only 1 node
       set_fact:
-        is_crc: "{{ True if 'crc' in nodes.metadata.labels[\"kubernetes.io/hostname\"] else False }}"
+        is_crc: "{{ True if 'crc' in node_info.resources.metadata.labels[\"kubernetes.io/hostname\"] else False }}"
```

Comment:
I didn't actually test single-node CRC detection, but I think this rescue block is no longer necessary because k8s_info returns an array of resources in either case.

Reply:
I don't remember this rescue block and why it was necessary. I'm not sure what `nodes` data looked like vs `k8s_info` and the difference between them on single vs multi-node deployment. I'm fine with removing if tested.

Reply:
The lookup would return an array if there were multiple, but just a single hash if there was one.

Reply:
Ahhhh ok, thanks!